app/coffee/CompileController.coffee

```diff
@@ -15,7 +15,10 @@ module.exports = CompileController =
 			ProjectPersistenceManager.markProjectAsJustAccessed request.project_id, (error) ->
 				return next(error) if error?
 				CompileManager.doCompile request, (error, outputFiles = []) ->
-					if error?.terminated
+					if error?.message is "invalid state"
```

Comment:
Can we create a new subclass of Error here, and do an `error instanceof InvalidStateError` check instead? That's a pattern I've been trying to push for when handling certain error cases, see e.g. https://github.com/sharelatex/web-sharelatex/blob/master/app/coffee/Features/Errors/Errors.coffee

Also, can we be a bit more descriptive than `InvalidState`? What about something like `FilesOutOfSync`?